src/main/java/net/earthcomputer/multiconnect/protocols/v1_12_2/mixin/MixinEntity.java

```diff
@@ -85,4 +90,11 @@ private void modifyFluidMovementBoundingBox(Tag<Fluid> fluidTag, CallbackInfoRet
         ci.setReturnValue(foundFluid);
     }
 
+    @Inject(method = "isTouchingWater", at = @At(value = "RETURN"), cancellable = true)
+    private void correctStatus(CallbackInfoReturnable<Boolean> cir) {
+        if (cir.getReturnValueZ() && !isSubmergedIn(FluidTags.WATER, true) && getWaterHeight() < 0.28888931871D) {
```

Comment:
This needs a check for whether the version is 1.12, so that it doesn't affect newer versions.